docs/usage.md

```diff
@@ -115,8 +219,45 @@ Calculates a weighted ratio based on the other ratio algorithms.
 ## process
 
 ### extract
+
 Find the best matches in a list of choices.
 
+Parameters:
+
+- **query**: *str*
+
+    String we want to find.
+
+- **choices**: *Iterable*
+
+    List of all strings the query should be compared with or dict with a mapping `{<result>: <string to compare>}`
+
+- **scorer**: *Callable*, default `fuzz.WRatio`
+
+    Optional callable that is used to calculate the matching score between
+    the query and each choice.
+
+- **processor**: *Callable*, default `utils.default_process`
+
+    Optional callable that reformats the strings. `utils.default_process`
+    is used by default, which lowercases the strings and trims whitespace
+
+- **limit**: *int*
+
+    Maximum amount of results to return.
+
+- **score_cutoff**: *float*, default `0`
+
+    Optional argument for a score threshold. Matches with
+    a lower score than this number will not be returned.
+
+Returns:
+
+- **matches**: *List[Tuple[str, float]]*
```

Comment:
The return type here should be Union[List[Tuple[str, float]], List[Tuple[str, float, str]]] (maybe it is more readable to write 
List[Tuple[str, float]] or List[Tuple[str, float, str]]), since it will return a list of (<string to compare>, <score>) for lists and (<string to compare>, <score>, <result>) for mappings. (Thats probably incorrect in the docstring aswell)

Reply:
Yeah, I just copied it from the docstrings.